spec/features/irs_attempts_api/event_tracking_spec.rb

```diff
@@ -31,11 +31,7 @@
 
     events = irs_attempts_api_tracked_events
 
-    expect(events.count).to eq(1)
-    event = events.first
-    expect(event.event_metadata[:email]).to eq(user.email)
-    expect(event.event_metadata[:success]).to eq(true)
-    expect(event.session_id).to eq('test-session-id')
+    expect(events.count).to be > 0
   end
 
```

Comment:
Does this change still accomplish what the test was put in place for? The `events` array may be greater than one as we continue to implement more events on sign-in.

Reply:
I would vote that we assert exactly which events we expect to see, and just update this as we go. This looks like it stops after sign-in so it shouldn't be _too_ bad. (Let me know if I'm being too optimistic.)

We could do something like:

```ruby
expected_events = ['email-and-password-submitted', 'otp-signin'] # I don't know what these values should actually be
expect(events.count). to eq expected_events.length
expect(events.collect{ |e| e.event_type }.to match_array(expected_events)
```

I don't like asserting count by itself (unless it's <2), because I've seen things like "expected 47 items but got 45" and _who knows_ what was missing. It becomes a real pain to debug. The `match_array` matcher will show us a diff of the array if it fails, which is way easier to debug, and it lets us assert exactly what events we get.

Arguably, we don't need to test the parameters of every event because those are separately being tested when we add each event. E.g., with multiple events, we don't need to have assertions around things like email address or success, just that the correct events were fired.

Reply:
How does the above look? This was my attempt at implementing a check for specific events safely.